### PR TITLE
[New Rule] Azure Automation Webhook Created

### LINKED ISSUE #110: [New Rule] Azure Automation Webhook Created
## Description

Azure Automation runbooks can be configured to execute via a webhook. A webhook is a custom URL passed to Azure Automation along with a data payload specific to the runbook. An adversary may create a webhook in order to trigger a runbook that contains malicious code.

```
event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:(MICROSOFT.AUTOMATION/AUTOMATIONACCOUNTS/WEBHOOKS/ACTION or MICROSOFT.AUTOMATION/AUTOMATIONACCOUNTS/WEBHOOKS/WRITE) and event.outcome:Success
```

## Required Info

- **Eventing Sources:**

`filebeat-*`

- **Target Operating Systems:**

- **Platforms**

Azure

- **Target ECS Version:** `1.5.0`
- **New fields required in ECS for this?** No
- **Related issues or PRs** None

## Optional Info
- **References:**
  - https://powerzure.readthedocs.io/en/latest/Functions/operational.html#create-backdoor
  - https://github.com/hausec/PowerZure
  - https://posts.specterops.io/attacking-azure-azure-ad-and-introducing-powerzure-ca70b330511a
  - https://www.ciraltos.com/webhooks-and-azure-automation-runbooks/

## Example Data

image

### PR DESCRIPTION
**Pull request by threat-punter**
## Issues

Resolves #110 

## Summary
Identifies when an Azure Automation webhook is created. Azure Automation runbooks can be configured to execute via a
webhook. A webhook is a custom URL passed to Azure Automation along with a data payload specific to the runbook. An
adversary may create a webhook in order to trigger a runbook that contains malicious code.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/azure/persistence_azure_automation_webhook_created.toml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+]
+risk_score = 21
+rule_id = "e9ff9c1c-fe36-4d0d-b3fd-9e0bf4853a62"
+severity = "low"
+tags = ["Elastic", "Azure", "Continuous Monitoring", "SecOps", "Configuration Audit"]
+to = "now-25m"
+type = "query"
+
+query = '''
+event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:(MICROSOFT.AUTOMATION/AUTOMATIONACCOUNTS/WEBHOOKS/ACTION or MICROSOFT.AUTOMATION/AUTOMATIONACCOUNTS/WEBHOOKS/WRITE) and event.outcome:Success
```
Comment (brokensound77) [inline rules/azure/persistence_azure_automation_webhook_created.toml:34]: does `MICROSOFT.AUTOMATION/AUTOMATIONACCOUNTS/WEBHOOKS/ACTION` log everytime a webhook is executed or only when it is created?

Comment (threat-punter) [reply rules/azure/persistence_azure_automation_webhook_created.toml:34]: Good question. I just reviewed the logs again and confirmed that no event was logged when I actually triggered the webhook.

Comment (brokensound77) [reply rules/azure/persistence_azure_automation_webhook_created.toml:34]: Awesome, thanks

File: rules/azure/persistence_azure_automation_webhook_created.toml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+creation_date = "2020/08/18"
+ecs_version = ["1.5.0"]
+maturity = "production"
+updated_date = "2020/08/18"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies when an Azure Automation webhook is created. Azure Automation runbooks can be configured to execute via a
+webhook. A webhook is a custom URL passed to Azure Automation along with a data payload specific to the runbook. An
```
Comment (brokensound77) [inline rules/azure/persistence_azure_automation_webhook_created.toml:11]:
```suggestion
webhook. A webhook uses a custom URL passed to Azure Automation along with a data payload specific to the runbook. An
```

File: rules/azure/persistence_azure_automation_webhook_created.toml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+[metadata]
+creation_date = "2020/08/18"
+ecs_version = ["1.5.0"]
```
Comment (peasead) [inline rules/azure/persistence_azure_automation_webhook_created.toml:3]:
```suggestion
ecs_version = ["1.6.0"]
```

**Pull request merged**